docs/docs/migrating-from-v1-to-v2.md

```diff
@@ -6,68 +6,62 @@ title: Migrating from v1 to v2
+- [Setting Up Dependencies](#setting-up-dependencies)
 
-_Tutorial:_ If you'd like a step-by-step guide, [follow the tutorial](/tutorial/) to get started with Gatsby v2.
+  - [Update Gatsby version](#update-gatsby-version)
+  - [Manually install React](#manually-install-react)
+  - [Manually install plugins’ peer dependencies](#manually-install-plugins'-peer-dependencies)
 
-_Starters:_ If you'd rather use one of the official starters, you're in luck, there's a v2 edition for each of them. Install your favourite one with the Gatsby CLI.
+- [Handling Breaking Changes](#handling-breaking-changes)
 
-`gatsby-starter-default` with v2:
+  - [Remove or refactor layout components](#remove-or-refactor-layout-components)
+  - [Change `navigateTo` to `push`](#change-navigateto-to-push)
+  - [Convert to either pure CommonJS or pure ES6](#convert-to-either-pure-commonjs-or-pure-es6)
+  - [Move `Babel Configuration`](#move-babel-configuration)
+  - [Manually specify PostCSS plugins](#manually-specify-postcss-plugins)
+  - [Don't query nodes by ID](#dont-query-nodes-by-id)
 
-```
-gatsby new my-default-project https://github.com/gatsbyjs/gatsby-starter-default#v2
-```
+- [Resolving Deprecations](#resolving-deprecations)
 
-`gatsby-starter-hello-world` with v2:
+  - [Import Link from Gatsby](#import-link-from-gatsby)
+  - [Import graphql from Gatsby](#import-graphql-from-gatsby)
+  - [Rename `boundActionCreators` to `actions`](#rename-boundactioncreators-to-actions)
+  - [Rename `pathContext` to `pageContext`](#rename-pathcontext-to-pagecontext)
+  - [Rename responsive image queries](#rename-responsive-image-queries)
+  - [`deleteNodes` API deprecated](#delete-nodes-api-deprecated)
 
-```
-gatsby new my-hello-world https://github.com/gatsbyjs/gatsby-starter-hello-world#v2
-```
+- [Other Changes Worth Noting](#other-changes-worth-noting)
 
-`gatsby-starter-blog` with v2:
+  - [Explicit query names no longer required](#explicit-query-names-no-longer-required)
+  - [Remove inlined CSS in `html.js`](#remove-inlined-css-in-htmljs)
+  - [Remove explicit polyfills](#remove-explicit-polyfills)
 
-```
-gatsby new my-blog https://github.com/gatsbyjs/gatsby-starter-blog#v2
-```
+- [For Plugin Maintainers](#for-plugin-maintainers)
 
-Read on for a detailed guide on what's new in version 2!
+  - [Setting the Proper Peer Dependencies](#setting-the-proper-peer-dependencies)
+  - [Change `modifyBabelrc` to `onCreateBabelConfig`](#change-modifybabelrc-to-oncreatebabelconfig)
+  - [Change `modifyWebpackConfig` to `onCreateWebpackConfig`](#change-modifywebpackconfig-to-oncreatewebpackconfig)
+  - [`createRemoteFileNode` API has changed](#createRemoteFileNode)
+  - [Only allow defined keys on node.internal object](#only-allow-defined-keys-on-the-node-internal-object)
+  - [Import `graphql` types from `gatsby/graphql`](#import-graphql-types-from-gatsbygraphql)
+  - [Plugin specific changes](#plugin-specific-changes)
```

Comment:
This is only item that doesn't fit exactly in category it was assigned to. Currently in "Plugin specific changes" there is 1 documented plugin change (`gatsby-plugin-typography`) that forces plugin **users** (not maintainers) to adjust some utility file in site code.

This potentially can be either moved out of general migration guide and be put this in plugin documentation or have better section name

Reply:
we could put that into changes worth noting section, or like you said move to docs for that plugin open to either option

Reply:
moved this to breaking changes as last item, since that's technically what it is